heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerMain.java

```diff
@@ -303,7 +303,7 @@ public boolean manageTopology() {
       // TODO(mfu): timeout should read from config
       SchedulerStateManagerAdaptor adaptor = new SchedulerStateManagerAdaptor(statemgr, 5000);
 
-      boolean isValid = validateRuntimeManage(adaptor, topologyName);
+      boolean isValid = (Command.KILL.equals(command)) || validateRuntimeManage(adaptor, topologyName);
```

Comment:
I agree. This will actually still fail if the scheduler state is missing which isn't great but that seemed to be a bit more to unravel but I can see if theres a way to succeed in that case as well.